packages/firestore/src/local/memory_persistence.ts

```diff
@@ -72,22 +75,27 @@ export class MemoryPersistence implements Persistence {
     return queue;
   }
 
-  getQueryCache(): QueryCache {
+  getQueryCache(): QueryCache<MemoryPersistenceTransaction> {
     return this.queryCache;
   }
 
-  getRemoteDocumentCache(): RemoteDocumentCache {
+  getRemoteDocumentCache(): RemoteDocumentCache<MemoryPersistenceTransaction> {
     return this.remoteDocumentCache;
   }
 
   runTransaction<T>(
     action: string,
-    operation: (transaction: PersistenceTransaction) => PersistencePromise<T>
+    operation: (
+      transaction: MemoryPersistenceTransaction
+    ) => PersistencePromise<T>
   ): Promise<T> {
     debug(LOG_TAG, 'Starting transaction:', action);
     return operation(new MemoryPersistenceTransaction()).toPromise();
   }
 }
 
-/** Dummy class since memory persistence doesn't actually use transactions. */
-class MemoryPersistenceTransaction implements PersistenceTransaction {}
+/**
+ * Memory persistence is not actually transactional, but future implementations
+ * may have transaction-scoped state.
+ */
+export class MemoryPersistenceTransaction implements PersistenceTransaction {}
```

Comment:
I would rename `MemoryPersistenceTransaction` to `MemoryTransaction`, which would match `IndexedDbTransaction`. Otherwise, maybe `MemoryPersistedTransaction`?

Reply:
Done. `MemoryTransaction`.